src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,148 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { div, h, h1, img } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, ButtonSecondary } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { MarkdownViewer, newWindowLinkRenderer } from 'src/components/markdown';
+import scienceBackground from 'src/images/science-background.jpg';
+import { Ajax } from 'src/libs/ajax';
+import { signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useOnMount } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState<boolean>();
+  const { signInStatus, termsOfService } = authStore.get(); // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
+  const acceptedLatestTos = signInStatus === 'signedIn' && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = signInStatus === 'signedIn' && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState<string>();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const acceptTosResponse = await Ajax().User.acceptTos();
+      const { enabled } = acceptTosResponse || { enabled: false };
```

Comment:
Added this conditional logic

Reply:
Why does `Ajax().User.acceptTos` swallow 404 errors and returned undefined?

Since this component reports an error if `enabled` is false, it seems like it would be simpler if `acceptTos` threw an error in that case and the error was handled in the `catch` block below.

Reply:
honestly that was smelly to me too but i was trying to keep the changes to a minimum, ill make that change 👍

Reply:
It seems like that might have been left over from a past migration: #2805.